### PR TITLE
Update local fonts query

### PR DESCRIPTION
**Pull request by LabhanshAgrawal**
The local fonts api has changed a bit
`navigator.fonts.query()` now returns a Promise for Array of font metadata instead of async iterator
See https://github.com/WICG/local-font-access#enumerating-local-fonts

### CODE REVIEW

File: addons/xterm-addon-ligatures/src/font.ts [hunk 1]
```diff
@@ -44,8 +44,8 @@ export default async function load(fontFamily: string, cacheSize: number): Promi
       }
       const fonts: Record<string, IFontMetadata[]> = {};
       try {
-        const fontsIterator: AsyncIterableIterator<IFontMetadata> = (navigator as any).fonts.query();
```
Comment (Tyriar) [inline addons/xterm-addon-ligatures/src/font.ts:47]: Let's move all the types to the top of the file so the API is all together:

```ts
interface IFontAccessNavigator {
  fonts: ...,
  permissions: ...
}
```

**Review by Tyriar — CHANGES_REQUESTED** [review]

**Review by Tyriar — APPROVED** [review]

**Comment by LabhanshAgrawal** [comment]
Can you please do a patch release of this?
Ligatures addon not working at all right now, without this.

**Comment by Tyriar** [comment]
@LabhanshAgrawal sorry, drowning in notifications. You can jump on the beta build if you ever need to get a change early. We'll probably release the new version next week some time.

**Comment by LabhanshAgrawal** [comment]
@Tyriar the latest release has ligatures addon 0.5.0 same as last release.

**Comment by Tyriar** [comment]
Updated: https://github.com/xtermjs/xterm.js/commit/094bcbd81d1f080d204331088aa99f4d7173d56c, release incoming

**Comment by LabhanshAgrawal** [comment]
@meganrogge @Tyriar any update? The release notes are updated but there's no new release on npm.

**Comment by Tyriar** [comment]
@LabhanshAgrawal the release build failed?

```
src/font.ts(40,31): error TS2352: Conversion of type 'Navigator' to type 'IFontAccessNavigator' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Property 'fonts' is missing in type 'Navigator' but required in type 'IFontAccessNavigator'.
```

https://dev.azure.com/xtermjs/xterm.js/_build/results?buildId=4348&view=logs&j=8d802004-fbbb-5f17-b73e-f23de0c1dec8&t=7a29959a-6c04-5e83-f8f0-319180787bbf

**Comment by LabhanshAgrawal** [comment]
Oh, strange that it didn't get detected in earlier ci runs
have opened #3371 for this

**Pull request merged**